CMakeLists.txt

```diff
@@ -551,17 +567,9 @@ if(NOT IOS)
 endif()
 
 if(NOT WIN32)
-	CHECK_SYMBOL_EXISTS(pthread_mutex_timedlock pthread.h HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL)
-	if (NOT HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL)
-		CHECK_LIBRARY_EXISTS(pthread pthread_mutex_timedlock "" HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB)
```

Comment:
The library checks are here, because there are buggy pthread libraries out there shipping the ```pthread_mutex_timedlock``` definition in the header but lacking an implementation.